source/Halibut/Transport/Proxy/HttpProxyClient.cs

```diff
@@ -431,12 +432,12 @@ void WaitForData(NetworkStream stream)
             }
         }
         
-        async Task WaitForDataAsync(NetworkStream stream, CancellationToken cancellationToken)
+        async Task WaitForDataAsync(NetworkTimeoutStream stream, CancellationToken cancellationToken)
```

Comment:
Now that we have cancellation and timeout on reads, this could probably be simplified to a read async with a short timeout set on the stream

Reply:
It looks like we **do** have test coverage in this area, so maybe we can do that instead 👍

Reply:
Scope hammer this for now